internal/provisioner/kops_utils.go

```diff
@@ -130,3 +132,35 @@ func updateWorkersKopsInstanceGroupValue(kops *kops.Cmd, kopsMetadata *model.Kop
 
 	return nil
 }
+
+// ModifyAMISuffix updates the AMI name based on the provided architecture label.
+// If the AMI is in "ami-*" format, it returns it unmodified.
+// If the AMI does not include an architecture suffix, it adds the default "amd64" or the provided arch label.
+// If the AMI includes a different architecture suffix than the label, it replaces the suffix with the label.
+func ModifyAMISuffix(ami, archLabel string) string {
+	if strings.HasPrefix(ami, "ami-") {
+		// Case 1: AMI starts with "ami-", return as is.
+		return ami
+	}
+
+	// Regex to find existing architecture suffix in the AMI value.
+	re := regexp.MustCompile(`-(amd64|arm64)$`)
+
+	if !re.MatchString(ami) {
+		// Case 2: No architecture suffix, append default "amd64" or the provided label.
+		defaultSuffix := "amd64"
+		if archLabel == "arm64" {
+			defaultSuffix = archLabel
+		}
+		return ami + "-" + defaultSuffix
```

Comment:
This is a little confusing to read as the default part my no longer be accurate. I would just call this variable `suffix`.

Reply:
changed upon suggestion